File: Examples/Python/generating-saving/without-cloud-storage/create-barcode-and-saveimage-to-localdisk.py
import asposebarcodecloud
from asposebarcodecloud.BarcodeApi import BarcodeApi
from asposebarcodecloud.BarcodeApi import ApiException

import ConfigParser

config = ConfigParser.ConfigParser()
config.readfp(open(r'../../data/config.properties'))
apiKey = config.get('AppConfig', 'api_key') 
appSid = config.get('AppConfig', 'app_sid') 
out_folder = config.get('AppConfig', 'out_folder')
data_folder = "../../data/" #resouece data folder

#ExStart:1
#Instantiate Aspose Barcode API SDK
api_client = asposebarcodecloud.ApiClient.ApiClient(apiKey, appSid, True)
barcodeApi = BarcodeApi(api_client);

#Set Filename of image
name = "sample-barcode"

#Set Text to encode inside barcode
text = "Aspose.BarCode"

#Set Barcode Symbology
type = "QR"

#Set Barcode Image Format
format = "PNG"

#Set Resolution along X and Y in dpi
resolutionX = 96.0
resolutionY = 96.0

#Set Width and Height of barcode unit
dimensionX = 0.7
dimensionY = 2.0

#Sets if checksum will be added to barcode image.
enableChecksum = "NO"

try:
            
    #invoke Aspose.BarCode Cloud SDK API to create barcode and get barcode image as a stream
    response = barcodeApi.GetBarcodeGenerate(text=text, type=type, format=format, resolutionX=resolutionX, resolutionY=resolutionY, dimensionX=dimensionX, dimensionY=dimensionY, enableChecksum=enableChecksum)
    
    if response.Status == "OK":
        #download generated barcode image from api response
        outfilename = out_folder + name + "." + format
        with open(outfilename, 'wb') as f:
                    for chunk in response.InputStream:
                        f.write(chunk)        
                        
except ApiException as ex:
            print "ApiException:"
            print "Code:" + str(ex.code)
            print "Message:" + ex.message    
#ExEnd:1    

File: Examples/Perl/generating-saving/without-cloud-storage/create-barcode-and-saveimage-to-localdisk.pl
use lib 'lib';
use strict;
use warnings;
use utf8;
use File::Slurp; # From CPAN
use JSON;

use AsposeBarCodeCloud::BarcodeApi;
use AsposeBarCodeCloud::ApiClient;
use AsposeBarCodeCloud::Configuration;

my $configFile = '../../data/config.json';
my $configPropsText = read_file($configFile);
my $configProps = decode_json($configPropsText);

my $data_path = '../../data/';
my $out_path = $configProps->{'out_folder'};

#ExStart:1
$AsposeBarCodeCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeBarCodeCloud::Configuration::api_key = $configProps->{'api_key'};
$AsposeBarCodeCloud::Configuration::debug = 1;

#Instantiate Aspose.BarCode API SDK
my $barcodeApi = AsposeBarCodeCloud::BarcodeApi->new();

#Set Filename of image
my $name = 'sample-barcode';

#Set Text to encode inside barcode
my $text = 'Aspose.BarCode for Cloud';

#Set Barcode Symbology
my $type = 'QR';

#Set Barcode Image Format
my $format = 'PNG';

#Set Resolution along X and Y in dpi
my $resolutionX = 96.0;
my $resolutionY = 96.0;

#Set Width and Height of barcode unit
my $dimensionX = 0.7;
my $dimensionY = 2.0;

#Sets if checksum will be added to barcode image
my $enableChecksum = 'NO';

#invoke Aspose.BarCode Cloud SDK API to create barcode and get barcode image as a stream                                   
my $response = $barcodeApi->GetBarcodeGenerate(name => $name, text => $text, type => $type, format => $format,
								resolutionX => $resolutionX, resolutionY => $resolutionY, dimensionX => $dimensionX, dimensionY => $dimensionY, enableChecksum => $enableChecksum);

if($response->{'Status'} eq 'OK'){
	#download barcode from api response
	my $output_file = $out_path . $name . '.' . $format;
	write_file($output_file, { binmode => ":raw" }, $response->{'Content'});
}
#ExEnd:1